x/rollapp/keeper/hard_fork.go

```diff
@@ -210,15 +215,16 @@ func (k Keeper) IsFirstHeightOfLatestFork(ctx sdk.Context, rollappId string, rev
 
 // is forking to the latest height going to violate assumptions?
 func (k Keeper) ForkLatestAllowed(ctx sdk.Context, rollapp string) bool {
-	lastHeight, ok := k.GetLatestHeight(ctx, rollapp)
-	if !ok {
-		return false
-	}
+	lastHeight, _ := k.GetLatestHeight(ctx, rollapp)
 	return k.ForkAllowed(ctx, rollapp, lastHeight)
 }
 
 // is the rollback fork going to violate assumptions?
 func (k Keeper) ForkAllowed(ctx sdk.Context, rollapp string, lastValidHeight uint64) bool {
 	ra := k.MustGetRollapp(ctx, rollapp)
-	return 0 < ra.GenesisState.TransferProofHeight && ra.GenesisState.TransferProofHeight <= lastValidHeight
+	// we can fork if
+	// - genesis bridge not opened
+	// or
+	// - genesis bridge opened and we don't rollback the past the proof height
+	return 0 == ra.GenesisState.TransferProofHeight || ra.GenesisState.TransferProofHeight <= lastValidHeight
```

Comment:
> 0 == ra.GenesisState.TransferProofHeight

Should extract this to method BridgeOpen 
Although I think something like this already exists